engine/gamesys/src/gamesys/scripts/script_particlefx.cpp

```diff
@@ -227,35 +227,57 @@ namespace dmGameSystem
      *
      * @name particlefx.stop
      * @param url [type:string|hash|url] the particle fx that should stop playing
+     * @param options [type:table] Options when stopping the particle fx. Supported options:
+     *
+     * - [type:boolean] `clear`: instantly clear spawned particles
+     *
      * @examples
      *
-     * How to stop a particle fx when a game object is deleted:
+     * How to stop a particle fx when a game object is deleted and immediately also clear
+     * any spawned particles:
      *
      * ```lua
      * function final(self)
-     *     particlefx.stop("#particlefx")
+     *     particlefx.stop("#particlefx", { clear = true })
      * end
      * ```
      */
     int ParticleFX_Stop(lua_State* L)
     {
-        int top = lua_gettop(L);
+        DM_LUA_STACK_CHECK(L, 0);
 
         dmGameObject::HInstance instance = CheckGoInstance(L);
 
-        if (top != 1)
-        {
-            return luaL_error(L, "particlefx.stop only takes a URL as parameter");
-        }
         dmGameSystemDDF::StopParticleFX msg;
         uint32_t msg_size = sizeof(dmGameSystemDDF::StopParticleFX);
 
         dmMessage::URL receiver;
         dmMessage::URL sender;
         dmScript::ResolveURL(L, 1, &receiver, &sender);
 
+        int clear_particles = 0;
+        if (!lua_isnil(L, 2)) {
+            luaL_checktype(L, 2, LUA_TTABLE);
+            lua_pushvalue(L, 2);
+            lua_pushnil(L);
+            while (lua_next(L, -2)) {
+                const char* option = lua_tostring(L, -2);
+                if (strcmp(option, "clear") == 0)
```

Comment:
Need help with option name:

* clear | clear_particles
* delete | delete_particles
* remove | remove_particles

?

Reply:
`clear` sounds good to me

Reply:
maybe `immediate_clear` if it's not too long

Reply:
Thanks. I think I'll stick with ’clear’. ’immediate_clear’ is too long I think.